settings/Mailing.setting.php

```diff
@@ -33,6 +33,28 @@
     'description' => ts('When CiviMail is enabled, users who "subscribe" to a group from a profile Group(s) checkbox will receive a confirmation email. They must respond (opt-in) before they are added to the group.'),
     'help_text' => NULL,
   ],
+  'no_reply_email_address' => [
+    'group_name' => 'Mailing Preferences',
+    'group' => 'mailing',
+    'name' => 'no_reply_email_address',
+    'type' => 'String',
+    'quick_form_type' => 'Select',
+    'html_type' => 'select',
+    'html_attributes' => [
+      'class' => 'crm-select2',
+    ],
+    'default' => NULL,
+    'pseudoconstant' => [
+      'callback' => 'CRM_Admin_Form_Preferences_Mailing::getAvailableFromEmailAddresses',
+    ],
```

Comment:
The function you're calling here serves no purpose other than to return options from an option group. You can do that directly:

```suggestion
    'pseudoconstant' => [
      'optionGroupName' => 'from_email_address',
    ],
```